dpcpp/matrix/coo_kernels.dp.cpp

```diff
@@ -61,11 +70,311 @@ namespace dpcpp {
 namespace coo {
 
 
+constexpr int default_block_size = 256;
+constexpr int warps_in_block = 4;
+constexpr int spmv_block_size = warps_in_block * config::warp_size;
+
+
+namespace {
+
+
+/**
+ * The device function of COO spmv
+ *
+ * @param nnz  the number of nonzeros in the matrix
+ * @param num_lines  the maximum round of each warp
+ * @param val  the value array of the matrix
+ * @param col  the column index array of the matrix
+ * @param row  the row index array of the matrix
+ * @param b  the input dense vector
+ * @param b_stride  the stride of the input dense vector
+ * @param c  the output dense vector
+ * @param c_stride  the stride of the output dense vector
+ * @param scale  the function on the added value
+ *
+ * @tparam ValueType  type of values stored in the matrix
+ * @tparam IndexType  type of matrix indexes stored in the structure
+ * @tparam Closure  type of the function used to write the result
+ */
+template <int subwarp_size = config::warp_size, typename ValueType,
```

Comment:
For all of these, I think it would be better in terms of clarity if it was either subgroup_size or warp_size.

```suggestion
template <int subgroup_size = config::warp_size, typename ValueType,
```